src/nodes.coffee

```diff
@@ -4756,13 +4791,18 @@ exports.StringWithInterpolations = class StringWithInterpolations extends Base
             comment.newLine = yes
           attachCommentsToNode salvagedComments, node
         if (unwrapped = node.expression?.unwrapAll()) instanceof PassthroughLiteral and unwrapped.generated and not @jsx
-          commentPlaceholder = new StringLiteral('').withLocationDataFrom node
-          commentPlaceholder.comments = unwrapped.comments
-          (commentPlaceholder.comments ?= []).push node.comments... if node.comments
-          elements.push new Value commentPlaceholder
+          if o.compiling
+            commentPlaceholder = new StringLiteral('').withLocationDataFrom node
+            commentPlaceholder.comments = unwrapped.comments
+            (commentPlaceholder.comments ?= []).push node.comments... if node.comments
+            elements.push new Value commentPlaceholder
+          else
+            elements.push null
```

Comment:
Coffeescript supports empty interpolations (eg `"#{}"`) while JS does not

So I represented an empty interpolation as a `null` element in the `TemplateLiteral` AST `expressions` list (see examples in empty interpolation AST tests below)